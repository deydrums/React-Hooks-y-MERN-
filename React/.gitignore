# proyectos
/00-proyectos