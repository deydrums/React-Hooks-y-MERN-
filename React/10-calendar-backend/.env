PORT=80
DB_CNN=mongodb+srv://deydrums:7VaOymNqlPNdUn12@cluster0.id2bg.mongodb.net/CalendarApp
SECRET_JWT_SEED=Esto-Es-una-PalAbRa-S3kr2t4-j3j3j3