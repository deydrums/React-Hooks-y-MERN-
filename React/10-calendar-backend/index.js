const express = require('express');
require('dotenv').config();
//Ejecutar express
var app = express();

//Directorio publico
app.use(express.static('public'));

//Rutas
//Reescribir rutas
app.use('/api/auth', require('./routes/auth'));

//TODO: CRUD: Eventos

//Escuchar peticiones
app.listen(process.env.PORT,()=>{
    console.log(`Servidor corriendo en puerto ${process.env.PORT}`)
});