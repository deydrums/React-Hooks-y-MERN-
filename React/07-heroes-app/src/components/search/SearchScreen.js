import React from 'react';
import { heroes } from '../../data/heroes';
import { useForm } from '../../hooks/useForm';
import { HeroCard } from '../heroes/HeroCard';

export const SearchScreen = () => {

    const [formValues, handleInputChange, reset] = useForm({
        hero: ''
    });
    const {hero} = formValues;

    const heroesFiltered = heroes;

    const handleSearch = (e) => {
        e.preventDefault();
        console.log(hero);
        reset();
    }

    return (
        <div>
            <h1 className="title">Buscar HÃ©roe</h1>
            <hr></hr>
            <div className="container-heroe-search">
                <div className="container-search">
                    <h4>Buscar</h4>
                    <hr></hr>
                    <form
                        onSubmit = {handleSearch}
                    >
                        <input
                            type="text"
                            placeholder="Buscar heroe..."
                            className="form-control"
                            name = 'hero'
                            autoComplete="off"
                            value = {hero}
                            onChange={handleInputChange}
                        >
                        </input>
                        <button
                            type="submit"
                            className="btn m-1 btn-block btn-outline-primary"
                        >
                            Buscar
                        </button>
                    </form>
                </div>
                <div className="container-imgs-search">
                    <h4>Resultados</h4>
                    <hr/>
                    <div className="content-cards animate__animated animate__fadeIn">
                        {
                            heroesFiltered.map(hero=>(
                                <HeroCard
                                    key = {hero.id}
                                    {...hero}
                                />
                            ))
                        }
                    </div>
                </div>
            </div>
        </div>
    )
}
