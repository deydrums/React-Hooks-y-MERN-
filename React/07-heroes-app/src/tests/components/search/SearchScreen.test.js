import React from 'react'
import {mount} from 'enzyme'
import { MemoryRouter, Route } from 'react-router-dom';
import { SearchScreen } from '../../../components/search/SearchScreen';

describe('Pruebas en SearchScreen', () => {

    test('Debe de mostrarse correctamente con valores por defecto ', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries = {['/search']}>
                <Route path="/search" component={SearchScreen}/>
            </MemoryRouter>
        );
        expect(wrapper.find('.alert-info').text().trim()).toBe('Buscar un heroe...');
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe de mostrar a Iron Man y el input con el valor del queryString ', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries = {['/search?q=iron%20man']}>
                <Route path="/search" component={SearchScreen}/>
            </MemoryRouter>
        );
        expect(wrapper.find('input').prop('value')).toBe('iron man');
        expect(wrapper.find('HeroCard').exists()).toBe(true);
    });
})
