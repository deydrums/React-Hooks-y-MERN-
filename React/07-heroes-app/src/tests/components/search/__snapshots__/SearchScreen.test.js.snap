// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en SearchScreen Debe de mostrarse correctamente con valores por defecto  1`] = `
<div>
  <h1
    className="title"
  >
    Buscar HÃ©roe
  </h1>
  <hr />
  <div
    className="container-heroe-search"
  >
    <div
      className="container-search"
    >
      <h4>
        Buscar
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="hero"
          onChange={[Function]}
          placeholder="Buscar heroe..."
          type="text"
          value=""
        />
        <button
          className="btn mt-3 btn-block btn-outline-primary button-search"
          type="submit"
        >
          Buscar
        </button>
      </form>
    </div>
    <div
      className="container-imgs-search"
    >
      <h4>
        Resultados
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Buscar un heroe...
      </div>
      <div
        className="content-cards animate__animated animate__fadeIn"
      />
    </div>
  </div>
</div>
`;
